src/pluginscript_callbacks.lua

```diff
@@ -114,14 +123,28 @@ pluginscript_callbacks.script_init = wrap_callback(function(manifest, path, sour
 		return
 	end
 	lps_coroutine_pool:release(co)
+
 	if type(script) ~= 'table' then
 		api.godot_print_error('Script must return a table', path, path, -1)
 		return
 	end
 
+    if is_yue then
+        if script.__base ~= nil then
```

Comment:
In case we detect a `.yue` file, we check if it exports a YueScript class instead of a table.
When it does we change the `script` to `script.__base` so it can detect the methods and variables of the class.

Reply:
I don't quite like these `if is_yue` branches, I think it's better to separate the code that parses Lua code and Yuescript code and get the script definition table. The import code below could still be the same for both languages, extracted to a new function that receives the input table and script manifest to fill.